TelemetryManager/Src/TMCircularBuffer.cpp

```diff
@@ -1,30 +1,56 @@
 
-#include "CircularBuffer.hpp"
+#include "TMCircularBuffer.hpp"
 
-TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(buf, size) {
+TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(*buf, size) {
     // Constructor
 }
 
-TMCircularBuffer::TMCircularBuffer(CircularBuffer buf): CircularBuffer(buf) {
+TMCircularBuffer::TMCircularBuffer(CircularBuffer* buf): CircularBuffer(*buf) {
     // Constructor
 }
 
 TMCircularBuffer::~TMCircularBuffer() {
     // Destructor
+   
+ }
+
+TMCircularBuffer::MAVLinkByte  TMCircularBuffer::dequeue() {
+    uint8_t* res;
+    if(read(res,1)){
+        return *res;
+    }
+    return *res;
```

Comment:
return data type inconsistent with a function declaration. `uint8_t` and `MAVLinkByte`

Reply:
Shouldn't have line 22 return *res since you failed to read from circular buffer, probably return `0` instead.